libs/base/System/File.idr

```diff
@@ -17,26 +17,71 @@ support fn = "C:" ++ fn ++ ", libidris2_support"
 libc : String -> String
 libc fn = "C:" ++ fn ++ ", libc 6"
 
+js_try_catch_lasterr_Int : String -> String
+js_try_catch_lasterr_Int x = "{try{" ++ x ++ ";return 0n}catch(e){process.__lasterr = e; return 1n}}"
+
+js_try_catch_lasterr_Ptr : String -> String
+js_try_catch_lasterr_Ptr x = "{try{" ++ x ++ "}catch(e){process.__lasterr = e; return null}}"
+
+js_open_file : String
+js_open_file = "return {fd:__require_fs.openSync(n, m), buffer: Buffer.alloc(0), name:n, eof: false}"
+
 %foreign support "idris2_openFile"
+         ("node:lambdaRequire:fs:(n, m) =>" ++ js_try_catch_lasterr_Ptr js_open_file)
 prim__open : String -> String -> Int -> PrimIO FilePtr
+
 %foreign support "idris2_closeFile"
+         ("node:lambdaRequire:fs:(fp) => __require_fs.closeSync(fp.fd)")
 prim__close : FilePtr -> PrimIO ()
 
 %foreign support "idris2_fileError"
+         "node:lambda:x=>(x===1n?BigInt(1):BigInt(0))"
 prim_error : FilePtr -> PrimIO Int
+
 %foreign support "idris2_fileErrno"
+         "node:lambda:()=>-BigInt(process.__lasterr.errno)"
 prim_fileErrno : PrimIO Int
 
+
+read_line_js : String
+read_line_js =
+   "(file_ptr =>{
```

Comment:
move to a separate module

camelcase names

Reply:
An alternative is to put functions such as these in a file in `support/js`.

Reply:
It'd probably be neater to put them in `support/js`, especially for consistency with the Scheme back ends

Reply:
> An alternative is to put functions such as these in a file in `support/js`.

I can go that way, I didn't went that way to avoid unnecessary definitions on the generated js. Should I do it?

Reply:
Yes, it will scale better than leaving javascript code scattered around the libs as string literals. If you are concerned about code bloat, you could put the functions in separate files.